### DISCUSSION TITLE
Mapping custom data points - LSC Smart Pet feeder

### DISCUSSION THREAD

**Discussion by capitar06** [make-all/tuya-local#2223] [Q&A]
Hi all,

Just started using TuyaLocal because of the huge support for devices, I used LocalTuya before but this Pet Smart Pet feeder had very limited entities/sensors etc. I added this pet feeder in TuyaLocal and got a lot more out of it, but I still miss some functions, like press to feed or a history of feeding.

Currently this feeder uses the ceres_plus_petfeeder.yaml from devices because that probably is the closest to that device. I got the DP_ids via the Iot platform from my own device, they are as followed:

`"result": {
    "properties": [
      {
        "code": "basic_flip",
        "custom_name": "",
        "dp_id": 103,
        "time": 1723910842385,
        "type": "bool",
        "value": false
      },
      {
        "code": "motion_sensitivity",
        "custom_name": "",
        "dp_id": 106,
        "time": 1723813952722,
        "type": "enum",
        "value": "0"
      },
      {
        "code": "basic_nightvision",
        "custom_name": "",
        "dp_id": 108,
        "time": 1723813952760,
        "type": "enum",
        "value": "0"
      },
      {
        "code": "sd_storge",
        "custom_name": "",
        "dp_id": 109,
        "time": 1723813953033,
        "type": "string",
        "value": "896|896|0"
      },
      {
        "code": "sd_status",
        "custom_name": "",
        "dp_id": 110,
        "time": 1723813953079,
        "type": "value",
        "value": 5
      },
      {
        "code": "sd_format",
        "custom_name": "",
        "dp_id": 111,
        "time": 1723650126400,
        "type": "bool",
        "value": false
      },
      {
        "code": "movement_detect_pic",
        "custom_name": "",
        "dp_id": 115,
        "time": 1723650126400,
        "type": "raw"
      },
      {
        "code": "sd_format_state",
        "custom_name": "",
        "dp_id": 117,
        "time": 1723650130139,
        "type": "value",
        "value": 0
      },
      {
        "code": "motion_switch",
        "custom_name": "",
        "dp_id": 134,
        "time": 1723650129819,
        "type": "bool",
        "value": false
      },
      {
        "code": "record_switch",
        "custom_name": "",
        "dp_id": 150,
        "time": 1723650130038,
        "type": "bool",
        "value": true
      },
      {
        "code": "record_mode",
        "custom_name": "",
        "dp_id": 151,
        "time": 1723803105627,
        "type": "enum",
        "value": "1"
      },
      {
        "code": "feed_num",
        "custom_name": "",
        "dp_id": 231,
        "time": 1723878182225,
        "type": "value",
        "value": 4
      },
      {
        "code": "food_weight",
        "custom_name": "",
        "dp_id": 232,
        "time": 1723650173063,
        "type": "value",
        "value": 10
      },
      {
        "code": "realtime_data",
        "custom_name": "",
        "dp_id": 233,
        "time": 1723878177015,
        "type": "enum",
        "value": "0"
      },
      {
        "code": "history_data",
        "custom_name": "",
        "dp_id": 234,
        "time": 1723650126400,
        "type": "bool",
        "value": false
      },
      {
        "code": "feed_voice_record",
        "custom_name": "",
        "dp_id": 235,
        "time": 1723650126400,
        "type": "enum",
        "value": "0"
      },
      {
        "code": "schedule",
        "custom_name": "",
        "dp_id": 236,
        "time": 1723877353330,
        "type": "string",
        "value": "7f0820041000000"
      }
    ]
  },
  "success": true,
  "t": 1723910852986,
  "tid": "cf9e5aaf5cb211ef883136d89ae34f8e"
}`

So here I see History data, Feed num which I think is feeding. 
I would like to add some of these DP_ids to enrich the entities/sensors/functions in HA for this device, but how do I go about this? I'm pretty new to this and can't really find any guides on how to actually do this. Most of them explain the proces of getting the device id, local key and DP_ids but don't actually show how I can map these.

If there is a guide or something available on how to map these DP_ids, that would be awesome!

**Comment by make-all** [comment]
History data is unlikely to be usefully available over the local protocol, most likely only the latest record is sent at end of cleaning, and the normal case is the cloud keeps the history and allows the app to query it, but this would not be applicable to local use.

**Reply by capitar06** [reply]
Thanks for the reply, that makes sense.

Other entities like:
ovement_detect_pic
motion_switch
feed_num
feed_num
schedule

Would also really come in handy as entities in HA so there is less reason to use the Smart Home app.

So my main question still remains how I can use a custom yaml to enrich HA entities.

**Comment by capitar06** [comment]
Still searching for a way to do this. The current entities/sensors seem to be a bit buggy sometimes, for example when enabling motion detection, it becomes unknown after 30 sec. So if there is someone who can help me out with using these DP's to enrich entities in HA, please let me know. 😄

**Reply by make-all** [reply]
Motion sensing should probably be an event, not a sensor, if the device is only sending the dp when there is motion detected..

**Comment by capitar06** [comment]
I mean for example here:
![image](https://github.com/user-attachments/assets/61ccb600-d122-49f2-8d02-a3ba5b0363e4)
When I press human detection is becomes unknown after a few sec:
![image](https://github.com/user-attachments/assets/771339a5-5c25-4cb8-9993-34a36df6486b)
Not sure how this happens. Same goes for motion tracking.

**Reply by make-all** [reply]
None of this matches the ceres_plus_petfeeder.yaml that you say you are using, and it also doesn't match the dps ids you listed above, so it is difficult to guess at what might be going wrong.

**Reply by capitar06** [reply]
> None of this matches the ceres_plus_petfeeder.yaml that you say you are using, and it also doesn't match the dps ids you listed above, so it is difficult to guess at what might be going wrong.

I ran another diagnostic and it seems to be using a different config now. I added the device again so maybe it choose another yaml this time. It seems to be using: lsc_ptz_camera
`"data": {
    "name": "Pet feeder",
    "type": "lsc_ptz_camera",
    "device_id": "**REDACTED**",
    "device_cid": "",
    "local_key": "**REDACTED**",
    "host": "**REDACTED**",
    "protocol_version": 3.2,
    "tinytuya_version": "1.15.1",
    "api_version_set": 3.2,
    "api_version_used": 3.2,
    "api_working": true,
    "status": {
      "103": null
`

Would this help? If there is anything I can do, please let me know. In the app I have the possibility to enable motion detection/sensitivity etc. so I think the device is capable.

I think the yaml lsc_ptz_camera is a security camera, which explains the issues and options to set cam PTZ which the pet feeder does not support.

Thanks.

**Reply by make-all** [reply]
The integration does not choose config files for you. One of the configuration steps is for you to choose a config file.  These petfeeder and camera devices are all rather erratic at providing the info in their dps, so they can only be matched based on one or two values, which causes many false positives. If the device is not a 100% match, it is likely it won't work perfectly. There is a warning message output when setting the device up, which tells how good a match it is.

**Reply by capitar06** [reply]
> The integration does not choose config files for you. One of the configuration steps is for you to choose a config file. These petfeeder and camera devices are all rather erratic at providing the info in their dps, so they can only be matched based on one or two values, which causes many false positives. If the device is not a 100% match, it is likely it won't work perfectly. There is a warning message output when setting the device up, which tells how good a match it is.

Correct, these are the given options when adding the device:
![image](https://github.com/user-attachments/assets/7db5ce4a-4845-4f9a-a0c9-d99fb49e543f)
So my guess is a camera or an aroma diffuser. So my main question still is, would it be possible to use the DP ids i retrieved from IoT platform? Or request a new yaml to be added with those DP ids? I know it's possible to do with local tuya but i have to add them manually per entitiy/sensor etc which is a bit of a hassle. 

Or if you have a another suggestion, please let me know.
